pkg/operator/ceph/controller/predicate.go

```diff
@@ -295,6 +295,55 @@ func WatchControllerPredicate() predicate.Funcs {
 	}
 }
 
+// WatchCephClusterPredicate is a predicate used by child controllers such as Filesystem or Object
+// It watch for CR changes on the CephCluster object and reconciles if this needs to be propagated
+// For instance the logCollector option from the CephCluster spec affects the configuration of rgw pods
+// So if it changes we must update the deployment
+func WatchCephClusterPredicate() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			logger.Debug("create event from a CR")
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			logger.Debug("delete event from a CR")
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			logger.Debug("update event from a CR")
+			// resource.Quantity has non-exportable fields, so we use its comparator method
+			resourceQtyComparer := cmp.Comparer(func(x, y resource.Quantity) bool { return x.Cmp(y) == 0 })
+
+			switch objOld := e.ObjectOld.(type) {
+			case *cephv1.CephCluster:
+				objNew := e.ObjectNew.(*cephv1.CephCluster)
+				logger.Debug("update event on CephCluster CR")
+				// If the labels "do_not_reconcile" is set on the object, let's not reconcile that request
+				isDoNotReconcile := IsDoNotReconcile(objNew.GetLabels())
+				if isDoNotReconcile {
+					logger.Debugf("object %q matched on update but %q label is set, doing nothing", DoNotReconcileLabelName, objNew.Name)
+					return false
+				}
+				diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
+				if diff != "" {
+					// If the log collector setting changes let's reconcile the child controllers
+					if objOld.Spec.LogCollector != objNew.Spec.LogCollector {
```

Comment:
In addition to checking for the changing log collector setting, if we check for the changed ceph image, do we not need the `notifyChildControllerOfUpgrade()` method anymore?

Reply:
Ah! This is sort of what I'm asking as part of my comment below: https://github.com/rook/rook/pull/7044#pullrequestreview-573626773

Reply:
Actually, the reason why I originally added `notifyChildControllerOfUpgrade()` was to start the child upgrade once the mon/mgr/osd are done upgrading. This is not really needed but recommended as part of the general upgrade procedure. Let's discuss this more.